### PR TITLE
feat: add Rust support in 'aspect init'

### PR DESCRIPTION
**Pull request by alexeagle**
Goals:
- should be as similar as possible to using Rust outside of Bazel. All non-Bazel tooling should continue to work.
- should be consistent with other languages under Aspect, for example the cargo "repinning" operation ought to appear similar-shaped

### CODE REVIEW

**Comment by alexeagle** [comment]
@illicitonion first thing I'm missing is: how do you run the `cargo` command that rules_rust has installed, so you can add a dependency? See npm, python, and Go where the README says how to do that: https://github.com/aspect-build/aspect-workflows-template/blob/main/%7B%7B%20.ProjectSnake%20%7D%7D/README.bazel.md#working-with-npm-packages

**Comment by illicitonion** [comment]
@alexeagle Ok, I've fixed up a few things, and got this working some - `cargo add` and friends are now supported and documented :) Shall we put together a list of things you'd be wanting to support out of the box?

I'm thinking (and please add/comment/edit):

### rustfmt via rules_lint

This _should_ Just Work, but we can do some testing and make sure the UX is as desired

### clippy via rules_lint
cc @mrmeku and https://github.com/aspect-build/rules_lint/issues/385

###  VS Code integration via rust-analyzer
If we're defaulting to "just lay stuff out like Cargo would expect", rust-analyzer should just work out of the box with nothing special.

But in general, rules_rust offers a target: `bazel run @rules_rust//tools/rust_analyzer:gen_rust_project` which will generate a `rust-project.json` in the workspace root (which we should probably `.gitignore`), which rust-analyzer will detect and configure itself with.

### protobuf
We have a choice to make here between using `build.rs` for cargo compatibility, or using `rules_proto` for re-use between other Bazel languages.

We could probably put together a "use rules_proto, and `copy_file` the generated code somewhere Cargo will see it" approach, and even if we wanted to be fancy generate a `build.rs` which runs Bazel to update this generated code...

**Comment by alexeagle** [comment]
> We should probably make the sample app be a clap application

We don't have any sample apps in this setup right now. It would be neat to scaffold those out but it should be a separate effort and include the existing languages

File: {{ .ProjectSnake }}/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fn main() {
```
Comment (alexeagle) [inline {{ .ProjectSnake }}/src/main.rs:1]: this file doesn't belong - we don't have "example application code" for the other languages.

If we wanted to add such a thing, we'd use https://hay-kot.github.io/scaffold/templates/scaffold-file.html#template-scaffolds - that would be cool but should be another PR I think

**Comment by alexeagle** [comment]
I forgot to say, yay, very cool :)

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -74,6 +75,28 @@ jobs:
               working-directory: "${{ steps.scaffold.outputs.dir }}"
               if: "${{ matrix.preset == 'js' }}"
 
+            - name: Rust smoke test
+              run: |
+                mkdir -p hello_world/src
+
+                cat >hello_world/BUILD.bazel <<EOF
+                load("@rules_rust//rust:defs.bzl", "rust_binary")
```
Comment (alexeagle) [inline .github/workflows/ci.yaml:83]: hm, we don't want to ask developers to learn writing BUILD files, is there no gazelle extension for Rust at all currently?
We should consider adding one soon, could use our starlark extension host ("Orion") - doesn't have to be in this PR.

Comment (thesayyn) [reply .github/workflows/ci.yaml:83]: I would love to write one!

Comment (alexeagle) [reply .github/workflows/ci.yaml:83]: Yeah we would make significant dogfood usage in Silo too, fun project when you get some time.

I can haz approve here so I can land?
